api/src/room_store.py

```diff
@@ -60,9 +62,30 @@ def __init__(self, path):
     def get_all_room_ids(self) -> list:
         return list(self.stored_data.keys())
 
-    def write_room_data(self, room_id: str, data: Iterable[Token]) -> None:
+    async def write_room_data(self, room_id: str, data: Iterable[Token]) -> None:
         storable_data = list(map(asdict, data))
         self.stored_data[room_id] = storable_data
 
-    def read_room_data(self, room_id: str) -> Optional[dict]:
+    async def read_room_data(self, room_id: str) -> Optional[dict]:
         return self.stored_data.get(room_id)
+
+
+class DatabaseRoomStore:
+    def __init__(self, path):
```

Comment:
That's fair; I was trying to stick to the same interface as the other room stores so that tests could still use the MemoryRoomStore. I'm not sure how db would play into that.

Reply:
don't need to match the interface for the constructor, as that's injected into the constructor of gss